1-js/05-data-types/02-number/article.md

```diff
@@ -331,35 +331,35 @@ There is a special built-in method [Object.is](mdn:js/Object/is) that compares v
 
 In all other cases, `Object.is(a, b)` is the same as `a === b`. 
 
-This way of comparison is often used in JavaScript specification. When an internal algorithm needs to compare two values for being exactly the same, it uses `Object.is` (internally called [SameValue](https://tc39.github.io/ecma262/#sec-samevalue)).
+This method of comparison is often used in JavaScript. When an internal algorithm needs to compare two values for being exactly the same, it uses `Object.is` (internally called [SameValue](https://tc39.github.io/ecma262/#sec-samevalue)).
 ```
 
 
 ## parseInt and parseFloat
 
-The numeric conversion using a plus `+` or `Number()` is strict. If a value is not exactly a number, it fails:
+Numeric conversion using a plus `+` or `Number()` is strict. If a value is not exactly a number, it fails:
 
 ```js run
 alert( +"100px" ); // NaN
 ```
 
-The sole exception is spaces before and after the line, they are ignored.
+The sole exception is spaces before and after the line, as they are ignored.
 
-But in real life we often have values in units, like `"100px"` or `"12pt"` in CSS. Also in many countries the currency symbol goes after the amount, so we have `"19€"` and would like to extract a numeric value out of that.
+But in real life we often have values in units, like `"100px"` or `"12pt"` in CSS. Also in many countries, the currency symbol goes after the amount, so we have `"19€"`, but would still like to extract a numeric value out of that.
 
 That's what `parseInt` and `parseFloat` are for.
 
-They "read" a number from a string until they can. In case of an error, the gathered number is returned. Function `parseInt` reads an integer number, `parseFloat` reads any number:
+They "read" numbers within a string and unless there's an error (for example, if the first character of a string is not a number), the first valid number is returned. The function `parseInt` returns an integer, whilst `parseFloat` will return a floating-point number:
```

Comment:
They read only *one* "number", not "numbers". The idea is that they aread from the start to the first error and then return what they've got.
Is there a better way to say that?

Reply:
Hahaha. Yea, this one is difficult to describe. I'll give it a shot, but I'm not sure if it will be adequate:

"They 'read' a string character by character, beginning with index 0. When they encounter the first non-numeral character, they stop reading and return the beginning portion of the string: the parsed number. If the first character is not a number, they return NaN."

That's a good start anyway :)

Reply:
Uhm yes, but `parseFloat` accepts the first point `.` and stops at the second one:
`parseFloat('1.2.3') == '1.2'`.

Reply:
Ahhhh. good _point_. I think I might have to recuse myself from this one. The topic is beyond my expertise.